### PR TITLE
doc: subsystem: settings: loading data behavior description

### PR DESCRIPTION
**Pull request by nvlsianpu**
It was unclear how loading data from persisted storage works.
This patch introduce explanation for that.

Code examples were aligned to recent changes.

Removed misleading information about simultaneous back-end support.

Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>

### CODE REVIEW

File: doc/subsystems/settings/settings.rst [hunk 1]
```diff
@@ -58,6 +57,17 @@ initializes the FCB area, so it must be called before calling
 ``settings_fcb_dst()``. File read target is registered using
 ``settings_file_src()``, and write target by using ``settings_file_dst()``.
 
+Loading data from persisted storage
+***********************************
+
+Call to ``settings_load()`` causes loading data from the storage to its volatile
```
Comment (dbkinder) [inline doc/subsystems/settings/settings.rst]: How about this:
```
A call to ``settings_load()`` uses an ``h_set`` implementation
to load settings data from storage to volatile memory.  The most
recent key values are guaranteed by traversing all stored content
and (potentially) overwriting older key values with newer ones.
After all data is loaded, the ``h_commit`` handler is issued,
signalling the application that the settings were successfully
retrieved.
```

File: doc/subsystems/settings/settings.rst [hunk 2]
```diff
@@ -43,8 +43,7 @@ These are registered using a call to ``settings_register()``.
 Persistence
 ***********
 
-Backend storage for the settings can be either FCB, a file in the filesystem,
-or both.
+Backend storage for the settings can be FCB or file in the filesystem.
```
Comment (dbkinder) [inline doc/subsystems/settings/settings.rst]:
```suggestion
Backend storage for the settings can be a Flash Circular
Buffer (FCB) or a file in the filesystem.
```

**Review by dbkinder — CHANGES_REQUESTED** [review]

File: doc/subsystems/settings/settings.rst (hunk 1 above)
Comment (Laczen) [inline doc/subsystems/settings/settings.rst]: The travelling through the persistent storage is dictated by the backend. If a backend implements the settings_load in a different way, repeated calls to h_set() could be avoided.
The h_commit() allows the settings routines to first retrieve settings from storage and then to apply them all at once.

Comment (nvlsianpu) [reply doc/subsystems/settings/settings.rst]: I made a note that it is for back-end we already have.

**Comment by nvlsianpu** [comment]
@Laczen , @dbkinder Pleas recheck

**Review by Laczen — APPROVED** [review]

**Review by dbkinder — APPROVED** [review]
doc changes look OK, thanks.

**Review by nashif — APPROVED** [review]

**Pull request merged**